### PR TITLE
Theming app support for chrome manifest.json file

### LINKED ISSUE #5371: manifest.json not using data from theme
### Steps to reproduce
1. Use a theme that has a custom title, name and logo.
2. Check the app manifest that is loaded from `/core/img/manifest.json`.

### Expected behaviour
The name and logo should be used from the theme.

### Actual behaviour
Name and logo are hardcoded to `Nextcloud` and the Nextcloud logo. I also tried putting a `manifest.json` in `core/img` of my theme. If the ` manifest.json` can be themed otherwise, I didn't find any documentation about it.

### Server configuration

**Nextcloud version:** (see Nextcloud admin page)
12.0.0

### PR DESCRIPTION
**Pull request by juliusknorr**
This PR will make the manifest.json file from #4091 being themed with the theming app values. It is also added on the login page now. I also moved the decision logic if an image should be overwritten by the theming app out of the URLGenerator, so we don't have more and more if statements in there in case we want to theme more images.

@nextcloud/theming 

Fixes #5371

### CODE REVIEW

File: apps/theming/lib/Controller/ThemingController.php [hunk 1]
```diff
@@ -423,4 +423,39 @@ public function getJavascript() {
...
+	/**
+	 * @NoCSRFRequired
+	 * @PublicPage
+	 *
+	 * @return Http\JSONResponse
+	 */
+	public function getManifest($app) {
+		$cacheBusterValue = $this->config->getAppValue('theming', 'cachebuster', '0');
+		$responseJS = [
+			"name" => $this->themingDefaults->getName(),
```
Comment (LukasReschke) [inline apps/theming/lib/Controller/ThemingController.php]: Single quotes pls :)

**Review by LukasReschke — CHANGES_REQUESTED** [review]

File: apps/theming/lib/Controller/ThemingController.php (hunk 1 above)
Comment (juliushaertl) [reply apps/theming/lib/Controller/ThemingController.php]: Sure. Fixed it in the other files as well. :wink:

**Review by LukasReschke — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Two tests are failing:

```
1) OCA\Theming\Tests\ThemingDefaultsTest::testReplaceImagePath with data set #2 ('core', 'favicon.ico')
Failed asserting that false matches expected 'themingRoute?v=0'.

/drone/src/github.com/nextcloud/server/apps/theming/tests/ThemingDefaultsTest.php:630

2) OCA\Theming\Tests\ThemingDefaultsTest::testReplaceImagePath with data set #3 ('core', 'favicon-touch.png')
Failed asserting that false matches expected 'themingRoute?v=0'.

/drone/src/github.com/nextcloud/server/apps/theming/tests/ThemingDefaultsTest.php:630
```

**Comment by MorrisJobke** [comment]
```
Analysing /drone/src/github.com/nextcloud/server/apps/theming/lib/ThemingDefaults.php
 1 errors
    line  272: OC_App - Static method of private class must not be called
```

File: core/templates/layout.guest.php [hunk 2]
```diff
@@ -13,6 +13,7 @@
 		<link rel="icon" href="<?php print_unescaped(image_path('', 'favicon.ico')); /* IE11+ supports png */ ?>">
 		<link rel="apple-touch-icon-precomposed" href="<?php print_unescaped(image_path('', 'favicon-touch.png')); ?>">
 		<link rel="mask-icon" sizes="any" href="<?php print_unescaped(image_path('', 'favicon-mask.svg')); ?>" color="<?php p($theme->getColorPrimary()); ?>">
+		<link rel="manifest" href="<?php print_unescaped(image_path('', 'manifest.json')); ?>">
```
Comment (skjnldsv) [inline core/templates/layout.guest.php]: Feels wrong to use a function named 'image_path' for a manifest, but if ti works.. :)
We should consider renaming it though!

**Review by skjnldsv — APPROVED** [review]
Ok here

File: core/templates/layout.guest.php (hunk 2 above)
Comment (juliushaertl) [reply core/templates/layout.guest.php]: Yeah, something like resource_path would make more sense indeed. But image_path is already used for the manifest file and also works with the old theme mechanism.

**Comment by MorrisJobke** [comment]
Still failing

File: core/templates/layout.guest.php (hunk 2 above)
Comment (skjnldsv) [reply core/templates/layout.guest.php]: That's why I approved anyway! :)

**Pull request merged**